src/Engine/Party.h

```diff
@@ -286,21 +286,38 @@ struct Party {
     /**
      * Return id of random character that can still act.
      *
-     * @return        ID of character or -1 if none of the character cat act.
+     * @param isGameRng   Use 'grng' if true, 'vrng' otherwise.
+     * @return            ID of character or -1 if none of the character cat act.
      */
-    int getRandomActiveCharacterId() const {
+    int getRandomActiveCharacterId(bool isGameRng) const {
         std::vector<int> activeCharacters = {};
         for (int i = 0; i < pPlayers.size(); i++) {
             if (pPlayers[i].CanAct()) {
                 activeCharacters.push_back(i);
             }
         }
         if (!activeCharacters.empty()) {
-            return activeCharacters[vrng->Random(activeCharacters.size())];
+            return activeCharacters[(isGameRng ? grng : vrng)->Random(activeCharacters.size())];
         }
         return -1;
     }
 
+    /**
+     * Return id of character that is represented by given pointer.
+     *
+     * @param player      Pointer to player class.
+     * @return            ID of character.
+     */
+    int getCharacterIdInParty(Player *player) {
```

Comment:
Similar function `Player::GetPlayerIndex()` should be dropped/ combined. Please add TODO